### PR TITLE
Add repository_scanning_configuration column to aws_ecr_repository table Closes #1716

### LINKED ISSUE #1716: Add new field in the aws_ecr_repository to show scanning configurations using batch-get-repository-scanning-configuration method
**Description**

In the `aws_ecr_repository` table, the image_scanning_configuration field is used to validate if ScanOnPush is enabled by accessing the JSON value that is called like this, but going deeper into this we saw that if we enable the Continuous Scanning option in private repositories all private repositories will have the option to scan _CONTINUOUS_ and the value of _ScanOnPush_ will continue with the previous state and now the value is scanning continues

**Describe the solution you'd like**
For that reason above, we believe that this `[batch-get-repository-scanning-configuration](https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_BatchGetRepositoryScanningConfiguration.html)` method returns the accurate data to show configurations about scanning, here an example of the method response:

```
failures: []
scanningConfigurations:
- appliedScanFilters:
  - filter: '*'
    filterType: WILDCARD
  repositoryArn: arn:aws:ecr:us-east-1:123456789:repository/scanning-on-push-disabled
  repositoryName: scanning-on-push-disabled
  scanFrequency: CONTINUOUS_SCAN
  scanOnPush: false
```

The solution is to create a field that populates the information the method above provides as a JSON field called `repository-scanning-configuration` due the data is more exact in terms to show if scanning is enabled and what type is configured.

**Describe alternatives you've considered**

**Additional context**
To enable continuos scanning in the private repositories, this as image reference for it 
Screenshot 2023-04-21 at 10 13 43

### PR DESCRIPTION
**Pull request by rajlearner17**
…ns using batch-get-repository-scanning-configuration method Closes #1716

# Integration test logs
<details>
  <summary>Logs</summary>

```
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select title, repository_scanning_configuration,image_scanning_configuration from aws_ecr_repository
[
 {
  "image_scanning_configuration": {
   "ScanOnPush": false
  },
  "repository_scanning_configuration": {
   "Failures": [],
   "ResultMetadata": {},
   "ScanningConfigurations": [
    {
     "AppliedScanFilters": [],
     "RepositoryArn": "arn:aws:ecr:us-east-1:698112345678:repository/demo",
     "RepositoryName": "demo",
     "ScanFrequency": "MANUAL",
     "ScanOnPush": false
    }
   ]
  },
  "title": "demo"
 }
]
```
</details>

### CODE REVIEW

**Comment by misraved** [comment]
@rajlearner17 could you please add an example query in the table docs using this particular column?

**Review by misraved — COMMENTED** [review]
@rajlearner17 please take a look at the minor review comments. Thanks!!

File: docs/tables/aws_ecr_repository.md [hunk 1]
```diff
@@ -127,3 +127,49 @@ where
...
+  repository_name,
+  r ->> 'AppliedScanFilters' as applied_scan_filters,
+  r ->> 'RepositoryArn' as repo_arn,
+  r ->> 'RepositoryName' as repo_name,
+  r ->> 'ScanFrequency' as scan_frequency,
+  r ->> 'ScanOnPush' as scan_push
+from
+  aws_ecr_repository,
+  jsonb_array_elements(repository_scanning_configuration -> 'ScanningConfigurations') as r
+
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply docs/tables/aws_ecr_repository.md:144]: @rajlearner17 

> @rajlearner17 please take a look at the minor review comments. Thanks!!

Could you please add a semicolon at the end of the every query?

File: docs/tables/aws_ecr_repository.md [hunk 2]
```diff
@@ -127,3 +127,49 @@ where
...
+```
+
+### List repository scan frequency is set to manual
+
+```sql
+select
+  repository_name,
+  r ->> 'RepositoryArn' as repo_arn,
+  r ->> 'RepositoryName' as repo_name,
```
[the remark this answers is not in the document]
Comment (rajlearner17) [reply docs/tables/aws_ecr_repository.md:153]: We don't need it again

**Review by ParthaI — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**